### PR TITLE
Parser: Fix - PAPP-23449 Run precommit and fix all lint issues

### PR DESCRIPTION
**Pull request by paulnawat**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [X] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [X] Bugfix
- [ ] Feature
- [X] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [X] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Update code to align with our new code style rules.

## What is the current behavior? (OPTIONAL)
- The pipeline is failing due to lint issue.

## What is the new behavior? (OPTIONAL)
- There should be no lint issue now.

---
Thanks for contributing!

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # File: __init__.py
 #
-# Copyright (c) 2017-2021 Splunk Inc.
+# Copyright (c) 2017-2022 Splunk Inc.
```
Comment (paulnawat) [inline __init__.py:3]: All Copyright years are manually updated. It's updated to 2022 b/c we're making change in this year!

File: parser.json [hunk 2]
```diff
@@ -9,12 +9,13 @@
     "product_name": "Parser",
     "product_version_regex": ".*",
     "publisher": "Splunk",
-    "license": "Copyright (c) 2017-2021 Splunk Inc.",
+    "fips_compliant": true,
```
Comment (paulnawat) [inline parser.json:12]: This is required to be set to pass the static test. Also, on the excel sheet, Parser is shown to be fips compliant on the excel sheet.

File: parser_connector.py [hunk 3]
```diff
@@ -15,21 +15,21 @@
...
+import calendar
 import email
+import json
+import sys
 import threading
-import parser_email
-import parser_methods
 import time
-import calendar
+
```
Comment (paulnawat) [inline parser_connector.py:24]: The changes in this file are:
* done as a result of running `pre-commit run --all-files`
* Manually update the too-long lines.

File: parser_email.py [hunk 4]
```diff
@@ -107,13 +105,23 @@
...
-IPV6_REGEX += r'|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|'
-IPV6_REGEX += r'(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|'
-IPV6_REGEX += r'(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|'
-IPV6_REGEX += r'(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|'
-IPV6_REGEX += r'(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|'
-IPV6_REGEX += r'(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?\s*'
+IPV6_REGEX += r'(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)'
```
Comment (paulnawat) [inline parser_email.py:108]: I simply break long lines into multiple lines to pass the lint issue.
See [reference](https://github.com/splunk-soar-connectors/office365/pull/6/files#diff-27e777a520241fb3d0c5b07591167393a460b00e8a3056045957a95782f8d63bR99)

File: parser.json (hunk 2 above)
Comment (paulnawat) [reply parser.json:12]: After doing global search for `hashlib.md5`, I don't see any `hashlib.md5` usage here.
See [reference](https://github.com/splunk-soar-connectors/crowdstrikeoauth/pull/1/files#diff-1ab4c4ae3c672ae5db676e2a88f429da50316741c9e072fc1a9673e9132f0ca8R278)

Comment (paulnawat) [reply parser.json:12]: The one hashlib usage is usee is `file_hash = hashlib.sha1(part_payload).hexdigest()` at https://github.com/splunk-soar-connectors/parser/blob/b50c7475faf41daaf1a3ae3a1a447e59007e56e3/parser_email.py#L673

Comment (paulnawat) [reply parser.json:12]: I believe it's fine to keep sha1 here.
cc: @chunmanjimmyf , @shiyingtu

File: parser_methods.py [hunk 5]
```diff
@@ -12,36 +12,35 @@
 # the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 # either express or implied. See the License for the specific language governing permissions
 # and limitations under the License.
-import sys
-import re
 import csv
+import re
```
Comment (paulnawat) [inline parser_methods.py:16]: Similar to other files, the change here is all refactoring from running `pre-commit run --all-files`. 
I have to manually update the too-long lines though.

File: parser_email.py (hunk 4 above)
Comment (rgil-splunk) [reply parser_email.py:108]: I know you didn't just add this, but this regex is HUGE 😱 I don't know where we got it from, but this seems error-prone

can we not just use the native python `ipaddress` library https://docs.python.org/3/library/ipaddress.html ? it'll automatically validate IPv4/IPv6 (and you can call just one of the specific subobjects when you need to differentiate during validation)

Comment (paulnawat) [reply parser_email.py:108]: @rgil-splunk I think that's a good point. Let's do this refactor in a separate ticket PAPP-23473.
This is to keep this CL simple and can be used as reference point later on (i.e. we can fall back to this CL, if the refactor goes wrong)
What do you think?

**Review by rgil-splunk — APPROVED** [review]

**Pull request merged**